flood_forecast/explain_model_output.py

```diff
@@ -1,167 +1,199 @@
-from typing import Dict
-import shap
-import torch
-from flood_forecast.preprocessing.pytorch_loaders import CSVTestLoader
+import random
 from datetime import datetime
+from typing import List, Optional
+
 import numpy as np
-import matplotlib.pyplot as plt
-import random
+import shap
+import torch
+
 import wandb
+from flood_forecast.named_dimension_array import NamedDimensionArray
+from flood_forecast.plot_functions import (
+    plot_shap_value_heatmaps,
+    plot_shap_values_from_history,
+    plot_summary_shap_values,
+    plot_summary_shap_values_over_time_series,
+)
+from flood_forecast.preprocessing.pytorch_loaders import CSVTestLoader
 
+BACKGROUND_BATCH_SIZE = 5
 
-BACKGROUND_SAMPLE_SIZE = 5
+
+def _prepare_background_batches(
+    model,
+    csv_test_loader: CSVTestLoader,
+    forecast_start_idx: int,
+    backgound_batch_size: int = BACKGROUND_BATCH_SIZE,
+) -> List[torch.Tensor]:
+    """Generate background batches for deep explainer.
+    Choose batches prior to forecast_start_idx as background data.
+    Each batch is of size (seq_len, num_feature)
+
+    Args:
+        model ([type]): trained model
+        csv_test_loader (CSVTestLoader): test data loader
+        forecast_start_idx (int): forecast start index
+        backgound_batch_size (int): number of batches used as background data 
+        for deep explainer. Default to BACKGROUND_BATCH_SIZE.
+
+    Returns:
+        List[torch.Tensor]: List of tensors of backgound_batch_size length
+    """
+    # select most recent 5 batches prior to forcast starte time as background
+    history_len = model.params["model_params"].get("seq_len")
+    if history_len is None:
+        history_len = model.params["dataset_params"]["forecast_history"]
```

Comment:
@michaelwang1994 I'm a bit hesitant extracting `history_len` in this way. Mainly because I'm not sure the difference between `seq_len` and `forecast_history` and where they are used and if there's conflict between these two.